token/transfer-hook-interface/src/inline_spl_token.rs

```diff
@@ -0,0 +1,27 @@
+//! Structs required to handle verify token mints. By having the types here, we
+//! don't need to create a circular dependency between token and permissioned-transfer
+
+use {
+    arrayref::{array_ref, array_refs},
+    solana_program::{program_error::ProgramError, program_option::COption, pubkey::Pubkey},
+};
+
+fn unpack_coption_key(src: &[u8; 36]) -> Result<COption<Pubkey>, ProgramError> {
+    let (tag, body) = array_refs![src, 4, 32];
+    match *tag {
+        [0, 0, 0, 0] => Ok(COption::None),
+        [1, 0, 0, 0] => Ok(COption::Some(Pubkey::new_from_array(*body))),
+        _ => Err(ProgramError::InvalidAccountData),
+    }
+}
+
+/// Extract the mint authority from the account bytes
+pub fn get_mint_authority(account_data: &[u8]) -> Result<COption<Pubkey>, ProgramError> {
+    const MINT_SIZE: usize = 82;
```

Comment:
Why not `use spl_token_2022::state::Mint::LEN as MINT_SIZE` ?

Reply:
Assuming the circular dep thing?

Reply:
Yeah exactly